CMakeLists.txt

```diff
@@ -4,21 +4,26 @@ project(nps_uw_sensors_gazebo)
 find_package(catkin REQUIRED tf)
 
 find_package(gazebo REQUIRED)
+find_package(roscpp REQUIRED)
+find_package(std_msgs REQUIRED)
 find_package(OpenCV)
+include_directories(${roscpp_INCLUDE_DIRS})
+include_directories(${std_msgs_INCLUDE_DIRS})
 include_directories(
   include
   ${catkin_INCLUDE_DIRS} 
   ${GAZEBO_INCLUDE_DIRS}
 )
 link_directories(${GAZEBO_LIBRARY_DIRS})
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GAZEBO_CXX_FLAGS}")
 
 catkin_package()
 
 ## Plugins
-add_library(nps_gazebo_ros_gpu_sonar_single_beam_plugin
-            src/nps_gazebo_ros_gpu_sonar_single_beam.cpp)
-target_link_libraries(nps_gazebo_ros_gpu_sonar_single_beam_plugin
-                      GpuRayPlugin ${catkin_LIBRARIES})
+add_library(gazebo_ros_pulse_lidar_plugin src/gazebo_ros_pulse_lidar.cpp)
```

Comment:
Library name conflict: Here we make libgazebo_ros_pulse_lidar_plugin.so but models/3dad_sl3/model.sdf wants libgazebo_ros_velodyne_gpu_laser.so.  To test, you might remove the old .so from uuv_ws/devel/lib.

Reply:
Bruce, we need both of those plugins. One is from the Gazebo library, and one is what I wrote. What error are you seeing?

Reply:
@M1chaelM , on "roslaunch nps_uw_sensors_gazebo uw_lidar_standalone.launch" Gazebo gives: "[Err] [Plugin.hh:178] Failed to load plugin libgazebo_ros_velodyne_gpu_laser.so: libgazebo_ros_velodyne_gpu_laser.so: cannot open shared object file: No such file or directory".

Reply:
Hm. I just ran catkin_make clean then catkin_make and launched it again with no error. Sorry to ask but can you check that you sourced the setup.bash?

Reply:
The plugin it is looking for should be at /opt/ros/melodic/lib/libgazebo_ros_velodyne_gpu_laser.so
Do you have it?

Reply:
Yes, I sourced setup.bash.  No, I don't have it.  I see, I need to install velodyne_gazebo_plugins...

Reply:
Is the installation for this plugin on the Wiki?  I don't see it at https://github.com/Field-Robotics-Lab/DAVE/wiki/Install-Directly-on-Host

Reply:
Oh, good to know. I'm not sure when that got installed.

Reply:
Does this work:
`sudo apt install ros-melodic-velodyne-gazebo-plugins`

Reply:
Yes, fixed.